### ISSUE TITLE
Configurable inter-track gap interval

### ISSUE THREAD

**Issue by rklrkl** [jwallet/spy-spotify#170]
**What problem will this new feature resolves ?**
Some albums/playlists have very short gaps between tracks (I have Crossfading and Automix disabled in the Spotify client to try to get a silence gap) and Spytify will often miss the gap and continue appending the next track onto the end of the current one leading to two (or more!) tracks combined into one mp3.

**What feature or enhancement would you like to see added ?**
It would like an option to specify the inter-track gap interval in 0.1 sec increments (min: 0.1 sec, maybe default 1.0 sec?) so that I can adjust this value for albums/playlists that have shorter than normal inter-track gaps. As it stands,  I have to create multiple 1-track playlists for adjacent tracks (and stop/start recording for each) with this problem, which is very inconvenient.

**Comment by jwallet** [comment]
i don't think this should be in any way a setting, but maybe a code fix. i will see how i could improve this. sometimes, it's just spotify that does not update its status at the proper time, resulting in faulty results with spytify. looks like they do it on purpose to mess with recorders.

**Comment by jwallet** [comment]
might be related to / fixed by #205 as well

**Comment by jmevel** [comment]
I too am facing this issue. 
@jwallet you can reproduce this bug with this album: https://open.spotify.com/album/3dPtahH1qEgcBi1Q7YwNSM?si=f17dMYWeTJmvenAEtvRpDg

**Comment by jwallet** [comment]
> I too am facing this issue.
> @jwallet you can reproduce this bug with this album: https://open.spotify.com/album/3dPtahH1qEgcBi1Q7YwNSM?si=f17dMYWeTJmvenAEtvRpDg

So after some tests, it looks like Spotify does not delivery the max quality of 160kbps for this album, it gets delivered (on my side) at 56kbps, mainstreams tracks on spotify are always at 160kbps (100%), so what could happen when encoding at lower bit rates is the length of the track increased without affecting the music, when playing the track it will stop playing the full track half way.

a track with an original length of 3 minutes, can gets 13 minutes length at lower bit rates, when you play it, it stops at 3 minutes / 13 minutes.

selected bit rate in spytify will affect encoding this way. Normally if you have Spotify Free you should select `160kbps` , Spotify Premium `320kpbs`. 

so on my side, I get the full track but with twice the length.

**Comment by alpharou** [comment]
Could it be possible to fetch the track metadata and then stop the recording exactly at the end? (Assuming the recording started exactly at 0:00).

Or hook to the windows media keys and hit rewind and play, then record at 0:00.

Then it would be possible to record crossfaded tracks.

**Comment by jwallet** [comment]
@alpharou this is something that won't work with lastfm api, and even when using spotify api it won't be perfect. 

upcoming spytify version will use spotiy api to generate track change events (currently the api is only used for fetching metadata), and even when using the api to detect when the track changes it can still have some delay. the api also differs from 1s to the streamable length of a track resulting in worst results (you don't want the next track to be recorded at the end of the current track for all your recordings).

the issue above is more likely to be caused by the pc going idle after the user left (more frequent on laptop than pc), you can install [caffeine](https://www.zhornsoftware.co.uk/caffeine/) to prevent your pc/laptop setting app process to sleep, (caffeine simulates key press every minute or so, which makes the pc thinks the user is still using the pc)

upcoming version will generate mp3 files will more accurate bit rate values, the quality should increased a bit and make stop #102 some bad encoding effects to happen.
